tests/amm_contracts/amm.rs

```diff
@@ -0,0 +1,542 @@
+#![deny(warnings)]
+
+use cairo_rs::vm::runners::cairo_runner::ExecutionResources;
+use felt::Felt;
+use num_traits::Zero;
+use starknet_rs::{
+    business_logic::{
+        execution::objects::{CallInfo, CallType},
+        fact_state::state::ExecutionResourcesManager,
+        state::state_api::StateReader,
+        transaction::error::TransactionError,
+    },
+    definitions::general_config::StarknetGeneralConfig,
+    services::api::contract_class::EntryPointType,
+    utils::Address,
+};
+use std::collections::HashSet;
+
+use crate::amm_contracts::utils::*;
+
+pub enum AmmEntryPoints {
```

Comment:
This enum and the From impl is not needed since we already have a function for getting the selector from the entrypoint name